### PR TITLE
Adds support for Google Maps basemaps

### PR DESCRIPTION
**Pull request by alonsogarciapablo**

### CODE REVIEW

File: src/geo/gmaps/gmaps-default-map-style.js [hunk 1]
```diff
@@ -1,69 +0,0 @@
-var DEFAULT_MAP_STYLE = [{
```
Comment (xavijam) [inline src/geo/gmaps/gmaps-default-map-style.js]: Why have you removed this? There will not be an option to create a google map instead a leaflet one?

Comment (alonsogarciapablo) [inline src/geo/gmaps/gmaps-default-map-style.js]: All of the Google Maps basemaps available in BUILDER have some styles, so there was no need to provide a default map style.

Comment (xavijam) [inline src/geo/gmaps/gmaps-default-map-style.js]: But I feel we should take into account when the user wants to create a map from the beginning using just the CARTO.js library, shouldn't we?

Comment (alonsogarciapablo) [inline src/geo/gmaps/gmaps-default-map-style.js]: That's something we'll have to think we work on the API. Anyway, if someone wanted to create a map from a custom viz.json for example, they [should be able to specify the `baseType` and `style` attributes](https://github.com/CartoDB/cartodb.js/blob/ff9d12ad6f78035d42dd224cc37c268346e7a159/src/geo/gmaps/gmaps-base-layer-view.js#L31-L40) to customise their maps. And, [they will get the default style if they use any of the default map types](https://github.com/CartoDB/cartodb.js/blob/ff9d12ad6f78035d42dd224cc37c268346e7a159/src/geo/gmaps/gmaps-base-layer-view.js#L6-L11).

**Comment by xavijam** [comment]
Hey, just found a small performance problem, in case we can fix it easily. Editing geometries with Google Maps as provider is not as smooth as in Leaflet. Here is an example (although a GIF is not the best example):

flickering-gmaps

Middle points flickers a lot, it is likely due to a performance problem, but it works 👌 .

**Comment by alonsogarciapablo** [comment]
@xavijam True fact. Performance is not very good when editing geometries, and it's indeed worst in the case of Google Maps. There's already an issue about this : https://github.com/CartoDB/cartodb.js/issues/1517. Perhaps we should prioritise it.
cc: @rochoa @donflopez

**Comment by xavijam** [comment]
That's your call guys, for me makes sense to tackle in a different issue, if not this is gonna be endless.

**Comment by alonsogarciapablo** [comment]
Yes, totally.

**Comment by xavijam** [comment]
Bugs fixed! 💃

**Pull request merged**